.github/workflows/CI-python.yml

```diff
@@ -74,15 +74,52 @@ jobs:
         # Use always() to always run this step to publish test results when there are test failures
         if: ${{ always() }}
 
-      - name: Upload coverage to Codecov
+      - name: Upload to codecov
+        id: codecovupload1
         uses: codecov/codecov-action@v2
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
           directory: ${{ matrix.packageDirectory }}
           env_vars: OS,PYTHON
-          fail_ci_if_error: true
+          fail_ci_if_error: false
           files: ./${{ matrix.packageDirectory }}/coverage.xml
           flags: unittests
           name: codecov-umbrella
           verbose: true
-        if: ${{ always() }}
+        if: matrix.pythonVersion == '3.7'
+
+      - name: echo status
+        if: always()
+        run: |
+          echo 'codecovupload1 job status: '
+          echo ${{ steps.codecovupload1.outcome }}
+          echo 'python version: '
+          echo ${{ matrix.pythonVersion }}
+          echo ${{ matrix.pythonVersion == '3.7' }}
+          echo ${{ matrix.pythonVersion }} == '3.7'
+          echo ${{ matrix.pythonVersion }} == 3.7
+          echo ${{ matrix.pythonVersion == 3.7 }}
+          echo matrix.pythonVersion == 3.7
```

Comment:
This is just for debugging and not meant to be checked in, right?

Reply:
yeah I am still debugging this, I can't seem to make the if statement work :(

Reply:
removed now